ops/framework.py

```diff
@@ -1051,12 +1053,27 @@ def __eq__(self, other):
             return self._under == other._under
         elif isinstance(other, collections.abc.Mapping):
             return self._under == other
+        elif isinstance(other, dict):
+            return dict(self._under) == other
         else:
             return NotImplemented
 
+    def to_json(self) -> str:
+        """Return a JSON representation of the underlying object."""
+        return json.dumps(dict(self._under), sort_keys=True)
+
     __repr__ = _wrapped_repr
 
 
+def _stored_dict_representer(dumper, data):
+    return dumper.represent_dict(
+        data._under
+    )
+
+
+yaml.add_representer(StoredDict, _stored_dict_representer)
```

Comment:
yaml.add_representer I feel is ok, as that is part of the design of the YAML library.
However, that doesn't mean we should be mutating upstream yaml's 'safe_load' function.